traffic_ops/traffic_ops_golang/deliveryservice/keys.go

```diff
@@ -210,9 +259,75 @@ func getSSLKeysByXMLIDHelper(xmlID string, inf *api.APIInfo, w http.ResponseWrit
 			return
 		}
 	}
+
 	api.WriteResp(w, r, keyObj)
 }
 
+// GetSSLKeysByXMLID fetches the deliveryservice ssl keys by the specified xmlID. V14 includes expiration date.
+func GetSSLKeysByXMLIDV15(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"xmlid"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	if inf.Config.RiakEnabled == false {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusServiceUnavailable, errors.New("the Riak service is unavailable"), errors.New("getting SSL keys from Riak by xml id: Riak is not configured"))
+		return
+	}
+	xmlID := inf.Params["xmlid"]
+	getSSLKeysByXMLIDHelperV15(xmlID, inf, w, r)
+}
+
+func getSSLKeysByXMLIDHelperV15(xmlID string, inf *api.APIInfo, w http.ResponseWriter, r *http.Request) {
+	version := inf.Params["version"]
+	decode := inf.Params["decode"]
+	if userErr, sysErr, errCode := tenant.Check(inf.User, xmlID, inf.Tx.Tx); userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	keyObj, ok, err := riaksvc.GetDeliveryServiceSSLKeysObjV15(xmlID, version, inf.Tx.Tx, inf.Config.RiakAuthOptions, inf.Config.RiakPort)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting ssl keys: "+err.Error()))
+		return
+	}
+	if !ok {
+		api.WriteRespAlertObj(w, r, tc.InfoLevel, "no object found for the specified key", struct{}{})
```

Comment:
`info` is an improper alert level to use when reporting an error, should be `error` level instead. Also shouldn't respond with an error using the `200 OK` response code; use an appropriate code on the range [400, 500) instead - probably `404 Not Found` judging by the context.

Reply:
so this one was also copied from above but the above use has "// empty response object because Perl" . is that something we still need? or can i change both of these to be api.HandleErr instead?

Reply:
I'm not sure, I'll need to look into that. But `info` is still an improper level either way.

Reply:
fair. ill update the level and we can change it to an api.HandleErr later if needed